src/backend/executor/nodeDynamicBitmapHeapscan.c

```diff
@@ -170,16 +170,17 @@ initNextTableToScan(DynamicBitmapHeapScanState *node)
 		 * the new varnos correspond to
 		 */
 		node->lastRelOid = currentPartitionOid;
+		pfree(attMap);
 	}
 
 	/*
-	 * For the very first partition, the targetlist of planstate is set to null. So, we must
-	 * initialize quals and targetlist, regardless of remapping requirements. For later
-	 * partitions, we only initialize quals and targetlist if a column re-mapping is necessary.
+	 * For the very first partition, the targetlist of planstate is set to null.
+	 * So, we must initialize quals and targetlist.
 	 */
-	if (attMap || node->firstPartition)
+	if (node->firstPartition)
```

Comment:
If we believe that initialization of targetlist and quals on this stage is odd then we might move this to executor init routine as it's done for SeqScan and IndexScan.